### PR TITLE
Add nomod makefile step for k8s.io/apimachinery

### LINKED ISSUE #60: Building from src code-generation fails
# description:
building from src with the makefile, failed in 2 points at moment.
1) the go.sum failed saying the checksum was outdated. ( by this error i just removed the file, then worked)
2) the deepcopy failed ( i have installed last one)
i have tried this out in 2 differents OS-Leap15 Computers

Using $GOPATH
~/go/src/github.com/kubic-project/kubic-init> 

# howto
when building from source with 
```
make
>>> Updating deepcopy files in pkg/config/
sh -c GO111MODULE=off deepcopy-gen -O zz_generated.deepcopy -i ./... -h ../../hack/boilerplate.go.txt
F1108 22:18:13.821865    6534 deepcopy.go:866] Hit an unsupported type invalid type.
pkg/config/config.go:18: running "sh": exit status 255
make: *** [Makefile:103: pkg/config/zz_generated.deepcopy.go] Error 1

go version go1.11.2 linux/amd64

```

As stated in desc, i was thinking i had an unstable env. but i tried it with a new Env on other Computer and i get now same error

### PR DESCRIPTION
**Pull request by MalloZup**
first experience with `gomod` :full_moon_with_face:  ( no big need for lot of emoticons in this PR)

fix #60

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -91,6 +91,7 @@ print-version:
 $(DEEPCOPY_GENERATOR):
 	@echo ">>> Getting deepcopy-gen (for $(DEEPCOPY_GENERATOR))"
 	@$(GO_NOMOD) get k8s.io/code-generator/cmd/deepcopy-gen
+	@$(GO_NOMOD) get -u k8s.io/apimachinery
```
Comment (MalloZup) [inline Makefile]: i was tempted to add a new Target, like 

`update apimachinery` or `api k8s` but  i think is ok for now

**Comment by MalloZup** [comment]
I think in longterm we should disable the possiblity to `develop` in `$GOPATH`  so we have only 1 behaviour, what do you think?

**Comment by inercia** [comment]
> I think in longterm we should disable the possiblity to `develop` in `$GOPATH` ( the oldschool way) so we have only 1 behaviour, what do you think?

I think users should be able to develop in or out of the `GOPATH`. However, right now, we are forced to checkout code inside of the `GOPATH` because of `deepcopy-gen`.

File: Makefile (hunk 1 above)
Comment (inercia) [reply Makefile]: I would not use `-u` (as we do with `deepcopy-gen`)
```suggestion
	@$(GO_NOMOD) get k8s.io/apimachinery
```

**Comment by MalloZup** [comment]
@inercia yop, i agree. I hope it will be more stable the situation in term of other pkgs conforming to the new modules... :pray: :church:

**Comment by inercia** [comment]
Merged, thank @MalloZup.

**Comment by MalloZup** [comment]
@inercia thanks, no big pr here were i can be proud of , it was just a starter for getting warm :sun_with_face: :stars: :cat2:

**Pull request merged**